Makefile

```diff
@@ -247,134 +234,130 @@ instrumentation-test-lint: npm-installed
+
+.PHONY: test-app-image-jvm
+test-app-image-jvm: ## Build the JVM test application.
+	@$(call build_container_image,$(TEST_APP_JVM_IMAGE_REPOSITORY),$(TEST_APP_JVM_IMAGE_TAG),test-resources/jvm/spring-boot)
+
+.PHONY: test-app-image-nodejs
+test-app-image-nodejs: ## Build the Node.js test application.
+	@$(call build_container_image,$(TEST_APP_NODEJS_IMAGE_REPOSITORY),$(TEST_APP_NODEJS_IMAGE_TAG),test-resources/node.js/express)
+
+.PHONY: dash0-api-mock-image
+dash0-api-mock-image: ## Build the Dash0 API mock container image, which is used in end-to-end tests.
+	@$(call build_container_image,$(DASH0_API_MOCK_IMAGE_REPOSITORY),$(DASH0_API_MOCK_IMAGE_TAG),test/e2e/dash0-api-mock)
+
 ##@ Build
 
 .PHONY: build
-build: husky-setup-hooks manifests generate fmt vet ## Build manager binary.
+build: manifests generate fmt vet ## Build manager binary.
 	go build -o bin/manager cmd/main.go
 
-.PHONY: run
-run: manifests generate fmt vet ## Run a controller from your host.
-	go run ./cmd/main.go
-
-# If you wish to build the manager image targeting other platforms you can use the --platform flag.
-# (i.e. docker build --platform linux/arm64). However, you must enable docker buildKit for it.
-# More info: https://docs.docker.com/develop/develop-images/build_enhancements/
-.PHONY: docker-build
-docker-build: \
-  docker-build-controller \
-  docker-build-instrumentation \
-  docker-build-collector \
-  docker-build-config-reloader \
-  docker-build-filelog-offset-sync \
-  docker-build-filelog-offset-volume-ownership
+.PHONY: images
+images: \
+  image-controller \
+  image-instrumentation \
+  image-collector \
+  image-config-reloader \
+  image-filelog-offset-sync \
+  image-filelog-offset-volume-ownership ## Build all container images used by the operator. If IMAGE_PLATFORMS is set, it will be passed as --platform to the build.
 
 define build_container_image
 $(eval $@_IMAGE_REPOSITORY = $(1))
 $(eval $@_IMAGE_TAG = $(2))
 $(eval $@_CONTEXT = $(3))
 $(eval $@_DOCKERFILE = $(4))
-if [[ -n "$($@_IMAGE_REPOSITORY)" ]]; then                                                                \
-  if [[ "$($@_IMAGE_REPOSITORY)" = *"/"* && "$($@_IMAGE_REPOSITORY)" != *"/library/"* ]]; then                                                         \
-    echo "not rebuilding the image $($@_IMAGE_REPOSITORY), this looks like a remote image";               \
-  else                                                                                                    \
-    dockerfile=$($@_DOCKERFILE);                                                                          \
-    if [[ -z $$dockerfile ]]; then                                                                        \
-        dockerfile=$($@_CONTEXT)/Dockerfile;                                                              \
-    fi;                                                                                                   \
```

Comment:
This is the part of the PR to stop applying weird heuristics about whether to build an image or not.